website/src/components/SearchPage/fields/AutoCompleteField.tsx

```diff
@@ -59,106 +88,272 @@ export const AutoCompleteField = ({
-            <Combobox
-                immediate
-                value={fieldValue}
-                onChange={(value) => setSomeFieldValues([field.name, value ?? NULL_QUERY_VALUE])}
-            >
-                <div className='relative'>
-                    <ComboboxInput
-                        displayValue={(value: string | number | null) => (value === null ? '(blank)' : String(value))}
-                        onChange={(event) => setQuery(event.target.value)}
-                        onFocus={load}
-                        placeholder={field.displayName ?? field.name}
-                        as={CustomInput}
-                    />
-                    {((fieldValue !== '' && fieldValue !== undefined) || query !== '') && (
-                        <button
-                            className='absolute inset-y-0 right-8 flex items-center pr-2 h-5 top-4 bg-white rounded-sm'
-                            onClick={() => {
-                                setQuery('');
-                                setSomeFieldValues([field.name, '']);
-                            }}
-                            aria-label={`Clear ${field.displayName ?? field.name}`}
-                        >
-                            <svg className='w-5 h-5 text-gray-400' fill='currentColor' viewBox='0 0 20 20'>
-                                <path
-                                    fillRule='evenodd'
-                                    d='M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z'
-                                    clipRule='evenodd'
+            <div className='w-full'>
+                <Combobox
+                    immediate
+                    multiple={multiSelect}
+                    value={multiSelect ? multiSelectValue : fieldValue}
+                    onChange={handleChange}
+                >
+                    <div className='relative'>
+                        {multiSelect ? (
+                            <FloatingLabelContainer
+                                label={field.displayName ?? field.name}
+                                isFocused={isFocused}
+                                hasContent={selectedValues.size > 0 || query !== ''}
+                                className='pr-16'
+                                onClick={(e) => {
+                                    // Focus the input when clicking anywhere in the field
+                                    // unless clicking on a button or the input itself
+                                    const target = e.target as HTMLElement;
+                                    if (!target.closest('button') && !target.closest('input')) {
+                                        inputRef.current?.focus();
+                                    }
+                                }}
+                            >
```

Comment:
This is huge. 200 lines of JSX alone in this component. Is there a reasonable way to refactor this into something that is easier to understand?

Reply:
I'm also not sure whether it makes sense to have the single and multi select in a single component. It of course makes sense to reuse existing components, but maybe we could extract common parts and assemble them into two separate components?
It could potentially get rid of many if branches that distribute over this file.